src/js/view/controls/components/menu/menu.js

```diff
@@ -294,51 +342,56 @@ export default class Menu extends Events {
         menu.destroy();
     }
     open(evt) {
-        if (this.visible && !this.openMenus) {
+        const mainMenuVisible = this.mainMenu.visible;
+        let focusEl;
+        if (!this.items.length || this.children) {
             return;
         }
-        backButtonTarget = null;
-        let focusEl;
         if (this.isSubmenu) {
+            const sourceEvent = evt.sourceEvent || event;
+            const firstItem = this.topbar ? this.topbar.firstChild : this.items[0].el;
+            const lastItem = this.items[this.items.length - 1].el;
+            const isKeydown = sourceEvent.type === 'keydown';
+            const key = isKeydown && sourceEvent.key.replace(/(Arrow|ape)/, '');
+            const itemTarget = isKeydown && key === 'Up' ? lastItem : firstItem;
+            if (this.visible && !this.openMenus.length) {
+                if (this.items.length && isKeydown) {
```

Comment:
what does this logic enable within the menu?

Reply:
A different focus target depending on if the menu is already visible and it's clicked again.

Reply:
ah yeah, if we had more time the controller idea you had would be killer here, this logic can get v tough to follow with all thee branchiing paths 👍